src/main/java/seedu/ui/AddUI.java

```diff
@@ -0,0 +1,176 @@
+package seedu.ui;
+
+import seedu.comparator.ClassNumComparator;
+import seedu.module.Lesson;
+import seedu.module.Module;
+import seedu.module.Semester;
+import seedu.timetable.Timetable;
+import seedu.timetable.TimetableLesson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class AddUI {
+    private static final int BALANCE_ARRAY = 1;
+    private static final int SERIAL_STARTING = 1;
+    private static final int DIVIDER = 42;
+    private static final int GAP = 46;
+    private static final String SPACE = " ";
+    private static final String DIV = "|";
+    private static final String LECTURE = "Lecture";
+    private static final String TUTORIAL = "Tutorial";
+    private static final String LAB = "Laboratory";
+
+    public void getLessonDetails(Semester semesterData, Timetable timetable, Module module) {
```

Comment:
Perhaps you could abstract this method into many other smaller methods? Currently it is over 100 LoC long.